### DISCUSSION TITLE
TerserPlugin has a bad banner plugin support

### DISCUSSION THREAD

**Discussion by TheColdPot** [webpack/webpack#12269] [Q&A] [answered]
I want to add a banner to my project, and I add this snippet to my `webpack.config.js`
```js
plugins: [
   new webpack.BannerPlugin('pecis | pecis.js.org')
]
```
It looks like it works but the result is
```js
/*! For license please see pecis.min.js.LICENSE.txt */
```
A `pecis.min.js.LICENSE.txt` is appended to my directory, the content is
```txt
/*! pecis | pecis.js.org */
```
But when I remove `TerserPlugin`, is works as excepted.
How do I config to makes it add banner directly?

Excepted output:
```js
/*! pecis | pecis.js.org */
// code
```

**Comment by alexander-akait** [comment, marked as answer]
Yep, good catch, WIP on this, as workaround you can do this https://github.com/webpack-contrib/terser-webpack-plugin#regexp (just change regexp)

**Reply by Hypnosphi** [reply]
This would extract the banner comment, I think it should rather be preserved using `terserOptions.format.comments`
https://github.com/webpack-contrib/terser-webpack-plugin#preserve-comments
